runtime/src/bank.rs

```diff
@@ -844,6 +846,11 @@ pub struct Bank {
 
     /// Fee structure to use for assessing transaction fees.
     fee_structure: FeeStructure,
+
+    /// The unique identifier for the corresponding block for this bank.
+    /// None for banks that have not yet completed replay or for leader banks as we cannot populate block_id
+    /// until bankless leader. Can be computed directly from shreds without needing to execute transactions.
+    block_id: RwLock<Option<Hash>>,
```

Comment:
Haven't yet added this to the serialized format. This will be done as part of bankless leader in order to add `block_id` to `bank_hash` (see https://github.com/AshwinSekar/solana/commit/d351d1508d1a9e11a1960e61199f89d02b6da4b1)